pkg/gofr/external_db.go

```diff
@@ -33,16 +39,20 @@ func (a *App) AddFTP(fs file.FileSystemProvider) {
 }
 
 // AddPubSub sets the PubSub client in the app's container.
-func (a *App) AddPubSub(pubsub container.PubSubProvider) {
+func (a *App) AddPubSub(ctx context.Context, pubsub container.PubSubProvider) error {
```

Comment:
This would result in a breaking change - user would need to handle the errors on their own rather than gofr taking care of that.

Reply:
gofr isn't really "taking care of" anything, it just hides the error in logging messages..